File: Frontend/BattleSquidsFrontApp/src/app/services/board.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { UrlService } from './url.service';
import { Observable } from 'rxjs';
import { Board } from '../models/board'
import { map } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class BoardService {
  url: string;

  constructor(private http: HttpClient, private urlService: UrlService) {
    this.url = urlService.getUrl() + "/board";
   }

   getAllBoard(): Observable<Board[]>
   {
    return this.http.get(this.url).pipe(map(resp => resp as Board[]));
   }

   addBoard(Board: Board):  Observable<number>
   {
     return this.http.post(this.url, Board).pipe(map(resp => resp as number));
   }

   getBoardById(id: number): Observable<Board>
   {
     return this.http.get(this.url + "/" + id).pipe(map(resp => resp as Board));
   }

   updateBoard(Board: Board): Observable<object>
   {
     return this.http.put(this.url, Board).pipe();
   }

   deleteBoard(id: number): Observable<object>
   {
     return this.http.delete(this.url + "/" + id).pipe();
   }
}

File: Backend/QuickLearner/src/main/java/com/revature/services/BoardService.java
package com.revature.services;

import java.util.Set;

import com.revature.beans.Board;

public interface BoardService {
	public Integer addBoard(Board b);
	public Set<Board> getAllBoard();
	public Board getBoardById(Integer id);
	public Board getBoard1ByGameId(Integer id);
	public Board getBoard2ByGameId(Integer id);
	public void updateBoard(Board b);
	public void deleteBoard(Board b);
	
}
